packages/analytics-core/src/plugins/destination.ts

```diff
@@ -95,65 +96,46 @@ export class Destination implements DestinationPlugin {
     });
   }
 
-  scheduleTryable(list: Context[], shouldAddToQueue = false) {
-    list.forEach((context) => {
-      // Only need to concat the queue for the first time
-      if (shouldAddToQueue) {
-        this.queue = this.queue.concat(context);
-      }
-      if (context.timeout === 0) {
-        this.schedule(this.config.flushIntervalMillis);
-        return;
-      }
-
-      setTimeout(() => {
-        context.timeout = 0;
-        this.schedule(0);
-      }, context.timeout);
-    });
-  }
-
-  addToQueue(...list: Context[]) {
-    const tryable = this.getTryableList(list);
-    this.scheduleTryable(tryable, true);
-    this.saveEvents();
-  }
-
-  schedule(timeout: number) {
+  sendEventIfReady() {
     if (this.scheduled || this.config.offline) {
       return;
     }
+    // if batching enabled, check if min threshold met for batch size
+    if (this.queue.length >= this.config.flushQueueSize) {
+      void this.flush(true);
+    }
 
     this.scheduled = setTimeout(() => {
       void this.flush(true).then(() => {
         if (this.queue.length > 0) {
-          this.schedule(timeout);
+          this.sendEventIfReady();
         }
       });
-    }, timeout);
+    }, this.config.flushIntervalMillis);
+
+    return;
   }
 
+  // flush the queue
   async flush(useRetry = false) {
     // Skip flush if offline
     if (this.config.offline) {
       this.config.loggerProvider.debug('Skipping flush while offline.');
       return;
     }
 
-    const list: Context[] = [];
-    const later: Context[] = [];
-    this.queue.forEach((context) => (context.timeout === 0 ? list.push(context) : later.push(context)));
-
     if (this.scheduled) {
       clearTimeout(this.scheduled);
       this.scheduled = null;
     }
 
-    const batches = chunk(list, this.config.flushQueueSize);
-
-    await Promise.all(batches.map((batch) => this.send(batch, useRetry)));
+    this.filterTriableList();
+    if (this.queue.length === 0) {
+      return;
+    }
 
-    this.scheduleTryable(later);
+    const currentFlushQueue = this.queue.slice(0, this.config.flushQueueSize);
+    await this.send(currentFlushQueue, useRetry);
```

Comment:
`amplitude.flush()` now would only send `flushQueueSize` events.

Reply:
Yea, you are right. Put a for loop and await on this.send.

Reply:
Thanks Alyssa for the change at https://github.com/amplitude/Amplitude-TypeScript/pull/736/commits/ae271bdd9bbcfc736de0565a0545acadbaca4b06. But we might need a more complex logic to enforce order when flush().

Edge cases: default configuration. 60 events in storage and manually call flush()
- batch 1 has an identify event
- batch 2 has only normal events
Request 1 with batch 1 failed but request 2 with batch 2 succeeded. So batch 2 is deleted from the storage while batch 1 will retry next time. 

We might need to check each response to decide whether to continue.

Reply:
That makes sense. I made `send` function return a status, and based on that stop sending other batches if there is an issue. Please check. Thanks.